subgraphs/burrow/src/handlers/farm.ts

```diff
@@ -0,0 +1,135 @@
+  let farm_id_asset = farm_id_obj.get("Supplied");
+  let farmType = "DEPOSIT";
+  if (!farm_id_asset) {
+    farm_id_asset = farm_id_obj.get("Borrowed");
+    farmType = "BORROW";
+    if (!farm_id_asset) {
+      log.warning("handleAddAssetFarmReward() :: farm_id_asset not found", []);
+      return;
+    }
+  }
+  const farm = farm_id_asset.toString();
+
+  const reward_token_id = data.get("reward_token_id");
+  if (!reward_token_id) {
+    log.warning("handleAddAssetFarmReward() :: reward_token_id not found", []);
+    return;
+  }
+  const reward_token = reward_token_id.toString();
+
+  const new_reward_per_day_ = data.get("new_reward_per_day");
+  if (!new_reward_per_day_) {
+    log.warning(
+      "handleAddAssetFarmReward() :: new_reward_per_day not found",
+      []
+    );
+    return;
+  }
+  const new_reward_per_day = new_reward_per_day_.toString();
+
+  const new_booster_log_base = data.get("new_booster_log_base");
+  if (!new_booster_log_base) {
+    log.warning(
+      "handleAddAssetFarmReward() :: new_booster_log_base not found",
+      []
+    );
+    return;
+  }
+
+  const reward_amount = data.get("reward_amount");
+  if (!reward_amount) {
+    log.warning("handleAddAssetFarmReward() :: reward_amount not found", []);
+    return;
+  }
+
+  const rewardToken = getOrCreateRewardToken(reward_token, farmType);
+
+  const market = getOrCreateMarket(farm);
+  const reward_tokens = market.rewardTokens!;
+  // push if reward token does not exists, get index
+  let reward_token_index = reward_tokens.indexOf(rewardToken.id);
+  if (reward_token_index == -1) {
+    reward_tokens.push(rewardToken.id);
+    market.rewardTokens = reward_tokens;
+    reward_token_index = reward_tokens.length - 1;
+  }
+
+  // add reward_amount to _reward_remaining_amounts
+  const _reward_remaining_amounts = market._reward_remaining_amounts;
+  if (_reward_remaining_amounts.length <= reward_token_index) {
+    _reward_remaining_amounts.push(BigInt.fromString(reward_amount.toString()));
+  } else {
+    _reward_remaining_amounts[reward_token_index] =
+      market._reward_remaining_amounts[reward_token_index].plus(
+        BigInt.fromString(reward_amount.toString())
+      );
+  }
+  market._reward_remaining_amounts = _reward_remaining_amounts;
+
+  // rewardTokenEmissionsAmount
+  const rewardTokenEmissionsAmount = market.rewardTokenEmissionsAmount;
+  if (rewardTokenEmissionsAmount!.length <= reward_token_index) {
+    rewardTokenEmissionsAmount!.push(
+      BigInt.fromString(new_reward_per_day.toString())
+    );
+  } else {
+    rewardTokenEmissionsAmount![reward_token_index] = BigInt.fromString(
+      new_reward_per_day.toString()
+    );
+  }
+  market.rewardTokenEmissionsAmount = rewardTokenEmissionsAmount;
+
+  // rewardTokenEmissionsUSD
+  const rewardTokenEmissionsUSD = market.rewardTokenEmissionsUSD;
+  const token = getOrCreateToken(rewardToken.token);
+  let price = token.lastPriceUSD!;
+  if (price.equals(BD_ZERO)) {
```

Comment:
what is this?

Reply:
sometimes oracles gives null price for BRR, so we override its's price

Reply:
I mean if it is 0, it is 0. We shouldn't override to .001

Reply:
sure will do; just note that'd give rewardUSD as 0 for almost all records

Reply:
fixed